06_puppy/sahidakhtar1/main.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+)
+
+type Puppy struct {
+	ID                   uint
+	Breed, Colour, Value string
+}
+
+type Storer interface {
+	CreatePuppy(Puppy)
+	ReadPuppy(uint) Puppy
+	UpdatePuppy(uint, Puppy)
+	DeletePuppy(uint) bool
+}
+
+//Map Store Starts
+type MapStore struct {
+	m map[uint]Puppy
+}
+
+func newMapStore() *MapStore {
+	ms := MapStore{}
+	ms.m = make(map[uint]Puppy)
+	return &ms
+}
+func (ms *MapStore) CreatePuppy(p Puppy) {
+	_, ok := ms.m[p.ID]
+	if ok {
+		return
+	}
+	ms.m[p.ID] = p
+}
+func (ms *MapStore) ReadPuppy(id uint) Puppy {
+	return ms.m[id]
+}
+func (ms *MapStore) UpdatePuppy(id uint, p Puppy) {
+	ms.m[id] = p
```

Comment:
maybe add: `p.ID = id`

Reply:
As pointer is not used.. it won't update the Puppy. Here `Puppy{1,"Bulldog","White","100"}` will be updated against the `id` passed.